public/src/map.js

```diff
@@ -35,7 +52,36 @@ export async function initMap(apiKey) {
   });
 
   // Add the geocoder to the map
-  map.addControl(geocoder);
+  // map.addControl(areaGeocoder);
+  map.addControl(constituencyGeocoder);
+
+  function dummy() {
+    console.log('dummy');
+    return [];
+  }
+  
+  function localSearch(query) {
+    const matchingFeatures = [];
+    return fetch('https://opendata.arcgis.com/datasets/937997590f724a398ccc0100dbd9feee_0.geojson')
+      .then(res => res.json())
+      .then((data) => {
+          console.log(data);
+          data.features.forEach(feature => {
+              if (
+                feature.properties.pcon19nm
+                  .toLowerCase()
+                  .search(query.toLowerCase()) !== -1
+              ) {
+                feature.properties.geometry = [feature.properties.long, feature.properties.lat];
+                feature['place_name'] = `📍 ${feature.properties.pcon19nm}`;
+                feature['center'] = [feature.properties.long, feature.properties.lat];
+                matchingFeatures.push(feature);
+              }
+          });
+          return Promise.resolve(matchingFeatures);
```

Comment:
I believe you can just return `matchingFeatures` directly

Reply:
done!